streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/ThreadMetrics.java

```diff
@@ -42,154 +45,215 @@ private ThreadMetrics() {}
-        addAvgAndMaxToSensor(punctuateSensor, THREAD_LEVEL_GROUP, tagMap, PUNCTUATE_LATENCY);
-        addInvocationRateAndCountToSensor(
-            punctuateSensor,
-            THREAD_LEVEL_GROUP,
-            tagMap,
+    public static Sensor punctuateSensor(final String threadId,
+                                         final StreamsMetricsImpl streamsMetrics) {
+        return invocationRateAndCountAndAvgAndMaxLatencySensor(
+            threadId,
             PUNCTUATE,
+            PUNCTUATE_RATE_DESCRIPTION,
             PUNCTUATE_TOTAL_DESCRIPTION,
-            PUNCTUATE_RATE_DESCRIPTION
+            PUNCTUATE_AVG_LATENCY_DESCRIPTION,
+            PUNCTUATE_MAX_LATENCY_DESCRIPTION,
+            Sensor.RecordingLevel.INFO,
+            streamsMetrics
         );
-        return punctuateSensor;
     }
 
-    public static Sensor skipRecordSensor(final String threadId, final StreamsMetricsImpl streamsMetrics) {
-        final Sensor skippedRecordsSensor =
-            streamsMetrics.threadLevelSensor(threadId, SKIP_RECORD, Sensor.RecordingLevel.INFO);
+    public static Optional<Sensor> commitOverTasksSensor(final String threadId,
+                                                         final StreamsMetricsImpl streamsMetrics) {
+        if (streamsMetrics.version() == Version.FROM_100_TO_23) {
+            final Sensor commitOverTasksSensor =
+                streamsMetrics.threadLevelSensor(threadId, COMMIT, Sensor.RecordingLevel.DEBUG);
+            final Map<String, String> tagMap = streamsMetrics.taskLevelTagMap(threadId, ROLLUP_VALUE);
+            addAvgAndMaxToSensor(
+                commitOverTasksSensor,
+                TASK_LEVEL_GROUP,
+                tagMap,
+                COMMIT_LATENCY,
+                COMMIT_OVER_TASKS_AVG_LATENCY_DESCRIPTION,
+                COMMIT_OVER_TASKS_MAX_LATENCY_DESCRIPTION
+            );
+            addInvocationRateAndCountToSensor(
+                commitOverTasksSensor,
+                TASK_LEVEL_GROUP,
+                tagMap,
+                COMMIT,
+                COMMIT_OVER_TASKS_RATE_DESCRIPTION,
+                COMMIT_OVER_TASKS_TOTAL_DESCRIPTION
+            );
+            return Optional.of(commitOverTasksSensor);
+        }
+        return Optional.empty();
+    }
+
+    private static Sensor invocationRateAndCountSensor(final String threadId,
+                                                       final String metricName,
+                                                       final String descriptionOfRate,
+                                                       final String descriptionOfCount,
+                                                       final RecordingLevel recordingLevel,
+                                                       final StreamsMetricsImpl streamsMetrics) {
+        final Sensor sensor = streamsMetrics.threadLevelSensor(threadId, metricName, recordingLevel);
```

Comment:
We have two overloaded `addInvocationRateAndCountToSensor` in `StreamsMetricsImpl`, and I feel using the one that constructs description as

```
            "The average per-second number of " + operation,
            "The total number of " + operation
```

is sufficient. Why do we duplicate the const string here and use the other overload?